extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3PropagatorExtractor.java

```diff
@@ -63,12 +65,25 @@ static SpanContext buildSpanContext(String traceId, String spanId, String sample
       }
     }
 
+    private static boolean isHex(String value) {
+      try {
+        new BigInteger(value, 16);
```

Comment:
I am somewhat concerned about creating one extra object every time we want to check if traceId or spanId are valid

Reply:
Agree, looping through and checking the character range shouldn't be too much more complex.